orangecontrib/text/widgets/owsentimentanalysis.py

```diff
@@ -0,0 +1,95 @@
+import numpy as np
+from nltk.corpus import opinion_lexicon
+from PyQt5.QtWidgets import QApplication
+
+from Orange.data import Domain, ContinuousVariable
+from Orange.widgets import gui
+from Orange.widgets.settings import Setting
+from Orange.widgets.widget import OWWidget
+from orangecontrib.text import Corpus
+from orangecontrib.text.preprocess import WordPunctTokenizer
+
+
+class OWSentimentAnalysis(OWWidget):
+    name = "Sentiment Analysis"
+    description = "Predict sentiment from text."
+    icon = "icons/SentimentAnalysis"
+    priority = 30001
+
+    inputs = [("Corpus", Corpus, "set_corpus")]
+    outputs = [("Corpus", Corpus, )]
+
+    autocommit = Setting(True)
+    want_main_area = False
+
+    def __init__(self):
+        super().__init__()
+
+        self.corpus = None
+        self.tokens = None
+        self.positive = set(opinion_lexicon.positive())
+        self.negative = set(opinion_lexicon.negative())
```

Comment:
Unless you plan to change these two sets at some point in the widget instance, you can make them class attributes -- move them above `def __init__` and remove `self.`.